### PR TITLE
MlFinLab packages versions update

### LINKED ISSUE #190: Sklearn Future Warning
Update package to be inline with the following:
```
FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
```

### PR DESCRIPTION
**Pull request by PanPip**
# Description

In this PR:
- The packages that MlFinLab is using are updated to the latest versions:
  - pandas: 0.25.3 -> 1.0.4
  - numpy: 1.17.3 -> 1.18.5
  - scikit-learn: 0.21.3 -> 0.23.1
  - scipy: 1.3.1 -> 1.4.1
  - matplotlib: 3.1.1 -> 3.2.1
  - cvxpy: 1.0.25 -> 1.1.1
- Travis CI build process is updated:
  - Now pylint shows only important results output and raw metrics tables (no dependency tree and unneeded tables)
  - All Warnings and leaked Errors during unit tests were fixed
  - During the build, only the errors in tests are displayed without unnecessary prints from test functions
  - Previous unit tests + coverage procedure was replaced by only coverage since errors in unit tests will anyways show during coverage test
  - Build time decreased from 40 minutes to 24 minutes
- Some unit tests were adjusted to decrease testing time and adjust to new versions of packages
- Functions that use multiprocessing (labeling, sample weights, sb_bagging, structural breaks) now have an option to silence the report of progress on asynch jobs
- Documentation issues were fixed
- All sphinx warnings were fixed
- Small imbalance sample dataset added

Fixes #190, #197, #228, #249, #335, #356, #358, 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

The following changes were tested on all previous MlFinLab unit tests.

Unit test values were changed only if the method algorithms was changed in the used package (mostly scikit-learn's RandomForestClassifier and BaggingClassifier functions). Or if the test dataset was decreased to improve testing time.

**Test Configuration**:
* Windows 10
* PyCharm

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -44,7 +44,10 @@ disable=I,
         duplicate-code,
         superfluous-parens,
         abstract-class-little-used,
-        too-few-public-methods
+        too-few-public-methods,
+        RP0401,
```
Comment (Jackal08) [inline .pylintrc:48]: What are these codes and can we swap them for the more descriptive keywords?

File: .travis.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ git:
 # command to run scripts
 script:
   - pylint mlfinlab --rcfile=.pylintrc -f text
-  - python -m unittest discover
+  - "travis_wait 20 sleep 1200 &"
```
Comment (Jackal08) [inline .travis.yml:17]: Re our call, we need to correct this.

File: coverage [hunk 3]
```diff
@@ -3,13 +3,12 @@
 echo "----Running Code Coverage----"
 
 # Remove multiprocessing coverage files in case a previous combine wasn't performed
-rm .coverage.*
 rm -fR cover/
 # Remove the main coverage file (.coverage)
 coverage erase
 
 # Discover and run all tests
-coverage run --concurrency=multiprocessing ./scripts/run_tests.py discover -v
```
Comment (Jackal08) [inline coverage:12]: Can we then remove this file? `/scripts/run_tests.py`

File: requirements.txt [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+matplotlib==3.2.1
+pandas==1.0.4
 pylint==2.4.3
-scikit-learn==0.21.3
-scipy==1.3.1
+scikit-learn==0.23.1
+scipy==1.4.1
 sphinx==2.2.1 # needed for docs
 sphinx-rtd-theme==0.4.3 # needed for docs
 statsmodels==0.11.1
```
Comment (Jackal08) [inline requirements.txt:13]: Do we still need xmlrunner?

**Comment by Jackal08** [comment]
I just updated develope into this branch. It may break things if the new code added isn't compliant.

File: .pylintrc (hunk 1 above)
Comment (PanPip) [reply .pylintrc:48]: I haven't found a way to replace them with keywoeds, but added comments describing what each code means.

File: .travis.yml (hunk 2 above)
Comment (PanPip) [reply .travis.yml]: Done. Alternative methods didn't work, so I'm keeping the version without `travis_wait` and decreasing verbose outputs to console.

File: coverage (hunk 3 above)
Comment (PanPip) [reply coverage:12]: Yes, removed now.

File: requirements.txt (hunk 4 above)
Comment (PanPip) [reply requirements.txt:13]: Removed

**Comment by Jackal08** [comment]
Have approved this PR. Please fix the merge conflict then merge into develop and delete branch. 

Really happy with this work, thank you.

**Pull request merged**